oot.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/bash
+
+
+_self_dir=$(realpath "$(dirname "$0")")
+
+
+function usage {
+	echo 'USAGE:'
+	echo -e "\t$0 [FLAGS] <device> [<device>...]\t\t(Example for akatsuki: $0 akatsuki)"
+	echo
+	echo 'FLAGS:'
+	echo -e "\t$0 -c <compiler>\t\tSelect a compiler to use. Currently supported compilers are gcc, linaro_gcc and clang (defaults to clang)."
+	echo -e "\t$0 -f\t\t\tFlash the kernel after compiling (using fastboot)."
+	exit 1
+}
+
+
+while getopts ':c:f' OPT; do
+	case ${OPT} in
+	c)
+		if [[ -f "$_self_dir/compile_$OPTARG.sh" ]]; then
+			_compiler=${OPTARG}
+		else
+			echo "Compiler '${OPTARG}' unknown or not implemented"
+			exit 1
+		fi
+		;;
+	f)
+		_fastboot_flash=true
+		;;
+	\?)
+		echo "Invalid option: -${OPTARG}"
+		usage
+	esac
+done
+
+shift $(( OPTIND - 1 ))
+
+if (( "$#" == 0 )); then
+	usage
+fi
+
+# Default to clang
+if [[ -z "${_compiler}" ]]; then
+	_compiler=clang
+fi
+
+
+# Platform common
+BOARD_KERNEL_BASE=0x00000000
+BOARD_KERNEL_PAGESIZE=4096
+BOARD_KERNEL_TAGS_OFFSET=0x01E00000
+BOARD_RAMDISK_OFFSET=0x02000000
+
+for _device in "$@"; do
+	unset BOARD_KERNEL_CMDLINE _platform _has_dtbo _recovery_ramdisk _verity_file _verity_key_id
+
+	BOARD_KERNEL_CMDLINE+=" lpm_levels.sleep_disabled=1"
+	BOARD_KERNEL_CMDLINE+=" service_locator.enable=1"
+	BOARD_KERNEL_CMDLINE+=" androidboot.hardware=${_device}"
+
+	# Device specific
+	case ${_device} in
+	kirin|mermaid)
+		_platform=ganges
+		;;
+	discovery|pioneer|voyager)
+		_platform=nile
+		;;
+	bahamut|griffin)
+		_platform=kumano
+		;;
+	pdx201)
+		_platform=seine
+		;;
+	akari|akatsuki|apollo)
+		_platform=tama
+		;;
+	lilac|maple|poplar)
+		_platform=yoshino
+		;;
+	*)
+		echo "Device '${_device}' unknown or not implemented"
+		exit 1
+		;;
+	esac
+
+	# Platform specific
+	case ${_platform} in
+	nile|ganges|yoshino)
+		_verity_file=build/target/product/security/verity.x509.pem
+		_verity_key_id=$(openssl x509 -in $_verity_file -text | grep keyid | sed 's/://g' | tr -d '[:space:]' | tr '[:upper:]' '[:lower:]' | sed 's/keyid//g')
+
+		BOARD_KERNEL_CMDLINE+=" veritykeyid=id:$_verity_key_id"
+		#case ${_platform} in
+		#nile|ganges)
+		#	BOARD_KERNEL_CMDLINE+=" earlycon=msm_serial_dm,0xc170000"
+		#	;;
+		#yoshino)
+		#	BOARD_KERNEL_CMDLINE+=" earlycon=msm_serial_dm,0xc1b0000 restore_msm_uart=0x03404000"
+		#	;;
+		#esac
+		;;
+	kumano)
+		export _has_dtbo=true
```

Comment:
The only remaining `export` is a little stubborn :joy:

Reply:
Yep :/